site/content/entry/yuzu-progress-report-apr-2022/index.md

```diff
@@ -0,0 +1,333 @@
+
+With the rewrite, yuzu now has a single "default service thread" that is used for 99% of the service methods that are non-blocking.
+For the services that are time-sensitive and for those that need blocking, we still allow thread creation (e.g. Audio, BSD, FileSystem, nvdrv)
+
+This brings down the service thread count from double digits to single digits, thus improving stability and consistency - especially on systems with less cores.
+Users with 4 thread CPUs (either 2 cores + HT/SMT, or 4 cores) should see performance and stability improvements on most games.
+
+Another battle for proper shutdown behaviour is fought and won.
+yuzu currently does not emulate multi-process capabilities of the HorizonOS kernel, however these still need to be managed.
+To begin, the HorizonOS services have a port (for both client and server) that is used as a channel of communication for multiprocess (between game process to the server process).
+A session is opened for each communication interface for them both and they are managed by their respective kernel objects.
+When the game closes the client port, the service closes the server port, and everything is shut down.
+
+The issue with our previous implementation was that yuzu wasn't properly tracking all the `KServerPort` and `KServerSession` objects for each service.
+And because of this, the services weren't properly getting closed and they in turn were causing further issues.
+
+This originally worked fine, but was regressed when we migrated guest kernel objects to emulated guest memory, as we mentioned previously.
+bunnei figured out the issue and quickly {{< gh-hovercard "8165" "reimplemented how we track these kernel objects" >}}.
+
+By having a single place where we can register/unregister open ports and sessions, we can now keep better track of these kernel objects.
+And by ensuring that they are closed when we tear down all services and kernel, we get much better emulation shutdown behaviour.
+
+## Input changes and general bugfixes
+
+If the user sets a very high DPI value for their mouse while using mouse panning, the cursor may be able to escape the rendering window.
+[IamSanjid](https://github.com/IamSanjid) implemented the {{< gh-hovercard "8170" "required fixes" >}}, including better centering timings to solve this issue. Thanks!
+
+[german77](https://github.com/german77) has several fixes ready for us.
+
+Let’s begin with an interesting one.
+yuzu’s screenshot capture feature allows an easy way to save moments at the resolution the scaler is currently set at.
+The hotkey for capture could be spammed, leading yuzu to a crash if several requests for capture were sent, this could be aggravated if the rendering resolution was set to a high value.
+To solve this, yuzu now {{< gh-hovercard "8192" "ignores new requests while a capture is being processed" >}}, and prints a warning in the log.
```

Comment:
```suggestion
The hotkey for screenshot capture could be spammed, leading yuzu to a crash if several requests for screenshots were sent. This could be worsened if the rendering resolution was set to a high value.
To solve this, yuzu now {{< gh-hovercard "8192" "ignores new requests while a capture is being processed" >}}, and prints a warning in the log.
```